### ISSUE TITLE
GDScript 2.0: `Array.pop_front()` causes error when result is a freed object

### ISSUE THREAD

**Issue by lapspider45** [godotengine/godot#62790]
### Godot version

v4.0.alpha11.official.afdae67cc

### System information

Linux

### Issue description

`Array.pop_front()` raises the error `Got a freed object as result of the call.` when the result is a freed object, pausing project execution. This doesn't happen if the result is discarded or if the array is explicitly/implicitly typed.

Probably affects at least `pop_back()` and `pop_at()` as well.

### Steps to reproduce

Create a node with the following script and at least 1 child.
```gdscript
extends Node2D

func _ready():
	var children = get_children() # returns an array
	children[0].free()
	var freed = children.pop_front() # this line errors out during runtime
```

**Comment by cdemirer** [comment]
The error is not specific to `Array` functions, it happens for all calls where the return value is not discarded (when debug is enabled).

```gdscript
func f():
	var o = MyClass.new()
	o.free()
	return o

func _ready() -> void:
	var t = f()
```

With a typed Array, it uses an optimized version of function call which apparently doesn't check for this. So, the bug in this issue seems to be the check not happening for typed Arrays.

**Comment by vonagam** [comment]
Should be already fixed by #76483. At least the snippet does not reproduce the error anymore.

**Comment by adamscott** [comment]
Closing as per @vonagam's comment.
